src/execute_lua_script.cc

```diff
@@ -79,4 +83,14 @@ execute_lua_script(sol::state& lua, sol::string_view script)
     }
 }
 
+gul14::expected<sol::load_result, std::string>
+load_lua_script(sol::state& lua, sol::string_view script)
+{
+    sol::load_result result = lua.load(script, anchor);
+    if (result.valid())
+        return result;
+
+    return gul14::unexpected(static_cast<sol::error>(result).what());
```

Comment:
Sol2 supports lots of weird casts of that type. From the docs:

> These functions load the desired blob of either code that is in a string, or code that comes from a filename, on the lua_State*. That blob will be turned into a Lua Function. It will not be run: it returns a load_result proxy that can be called to actually run the code, when you are ready. It can also be turned into a sol::function, a sol::protected_function, or some other abstraction that can serve to call the function.

What is going on under the covers there is beyond me... :shrug: